libswc/keyboard.c

```diff
@@ -191,7 +192,8 @@ keyboard_create(void)
 		goto error1;
 	}
 
-	if (!(xkb->keymap.map = xkb_keymap_new_from_names(xkb->context, NULL, 0))) {
+	if (!(xkb->keymap.map = xkb_keymap_new_from_names(xkb->context, names, 0)) &&
+	    !(xkb->keymap.map = xkb_keymap_new_from_names(xkb->context, NULL, 0))) {
```

Comment:
I don't quite understand the fallback here. Shouldn't it be an error if the right keymap can't be created?